### ISSUE TITLE
BpTreeNode의 맴버 함수 설계 문제

### ISSUE THREAD

**Issue by onting** [DSLDataStorage/DS_Project_2_2020_2#2]
BpTreeNode의 맴버 함수인 getIndexNode, getDataNode가 위험하게 설계된 것 같습니다. block scope안의 변수의 주소를 리턴하는 구문이 있습니다. 이것은 stack영역의 주소를 리턴하게 되어서 메모리 문제를 발생할 수 있습니다.
pure virtual function으로 바꿔야 합니다.  아니라면 만약에 의도한대로 BpTreeNode형의 객체에서 호출할 때 empty map의 주소를 리턴하게 하고 싶다면 smart pointer를 써서 함수 내에서 map 객체를 할당하고 리턴 타입을 shared pointer로 바꿔야합니다.

**Comment by DSLDataStorage** [comment]
이번 프로젝트의 목적은 상속이 아닌 자료구조입니다. 상속에 대한 문제로 혼란을 줄이기 위해 코드를 제공하였고 저 부분에 문제가 크게 있다고 학생이 판단이 되면 변경을 하셔도 상관은 없습니다.

**Comment by onting** [comment]
아니요, 모든 학생들이 수정된 스켈레톤 코드를 받아야 한다고 생각합니다. 문제가 있는 코드를 기반으로 작업을 하는 것은 문제가 있다고 생각하고, 학생들의 혼란을 막기위해서 문제가 있는 코드를 제공했다는 것 역시 학생들을 바보로 생각하는 것입니다. 왜 문제가 되는지 모르겠으면 아래의 사진을 참조하세요.
![image](https://user-images.githubusercontent.com/10670129/95942987-91556580-0e1f-11eb-87f2-b475b06b134a.png)

**Comment by DSLDataStorage** [comment]
프로젝트 구현을 하시다보면 아시겠지만, 해당 방법과 같은 호출로 선언하는 경우가 아니라 상속을 통해 오버라이딩으로 구현하게 됩니다. 그러므로, 위와 같은 경우는 발생하지 않습니다.  해당 코드가 불편하다면 공백으로 처리해서 구현하시면 됩니다.

**Comment by DSLDataStorage** [comment]
학생 의견을 반영하여 BpTreeNode.h 부분의 getIndexNode와 getDataNode의 반환을 return {};으로 변경하였으니 변경된 코드를 다시 받으세요
